### DISCUSSION TITLE
All three loading states remain to true even tho no fetch is currently active

### DISCUSSION THREAD

**Discussion by TitoSniberb** [TanStack/query#5943] [Q&A] [answered]
https://gyazo.com/cfa329dab622eb8131a36a44716bad6f

isLoading, isFetching and isInitialLoading, all remain to true when the query finished fetching (in my case with a code 500 received from backend).

Any ideas of what could be happening? This is the first time I have this problem and i've been using the library for almost a year now.
I can give more info but I dont even know what I can share that will be helpful as this doesnt make any sense. 
Version is 4.33

**Comment by frederikhors** [comment]
Can you share code? Are you using Svelte?

**Reply by TitoSniberb** [reply]
Im using react. And im 100% sure the call has ended, its not pending

```ts
export const useCalculateDistributionsQuery = (
  body: DistributionsRequestBody,
  additionalOptions?: { enabled?: boolean },
) => {
  const {
    data: distributions,
    isFetching,
    isLoading,
    isInitialLoading,
    ...rest
  } = useQuery({
    queryKey: ['calculateDistributions', body],
    queryFn: () => ApiAnalysisService.analysisService.calculateDistributions({}, body),
    ...noRefetch,
    retry: shouldRetryQuery,
    ...additionalOptions,
  });
  const isCalculatingDistributions = isFetching;

  return { distributions, isCalculatingDistributions, ...rest };
};
```

**Reply by frederikhors** [reply]
This is also happening with Svelte, I'll post an issue about it soon. I can't help you with React.

**Reply by TkDodo** [reply]
would be good if you could show a codesandbox reproduction please (preferably in react 😅)

**Reply by TkDodo** [reply]
also, instead of showing the flags, please show the values of `status` and `fetchStatus`. The flags are just derived state.

**Reply by TitoSniberb** [reply, marked as answer]
Sorry for the delay in the response. I found out that the problem was in the `retry` option. If you use a custom retry and any unhandled error happens inside that function, the props `status` and `fetchStatus` remain in loading and fetching

**Reply by TitoSniberb** [reply]
Shouldnt the status update to finished if any error occurs at any point? @TkDodo

**Reply by frederikhors** [reply]
Can you please post a reproduction using Stackblitz?

**Reply by TitoSniberb** [reply]
Im kinda in a hurry right now and can't do that, but I can easly reproduce it by just throwing a random error inside the retry, if it happens for react it should be happening for Svetle too
